### DISCUSSION TITLE
[Question] OnSubmit change button to progress bar

### DISCUSSION THREAD

**Discussion by ymoreiratiti** [vip-git/react-jsonschema-form-material-ui#474] [General]
It's possible to make onSubmit a promise, that when processing changes the button submit to a progress bar and disable click?

**Comment by vip-git** [comment]
Hi @ymoreiratiti - you can replace the onsubmit with your own ‘onSubmit’ handler that processes it as a promise. Not just that you can even replace the submit button itself with your own component.

you can look at this issue for example #449 where similar question was raised.

I hope this helps in resolving your issue. Thanks.
